cabal-install/Distribution/Client/GlobalFlags.hs

```diff
@@ -219,27 +216,32 @@ initSecureRepo :: Verbosity
                -> (SecureRepo -> IO a)  -- ^ Callback
                -> IO a
 initSecureRepo verbosity httpLib RemoteRepo{..} cachePath = \callback -> do
-    withRepo $ \r -> do
-      requiresBootstrap <- Sec.requiresBootstrap r
+    requiresBootstrap <- withRepo [] Sec.requiresBootstrap
+
+    mirrors <- if requiresBootstrap
```

Comment:
I considered that, but I didn't want to have to duplicate the detection in `queryBootstrapMirrors` for when a DNS query is going to occur at all... 

So what about:

```hs
if requiresBootstrap
then do
  info verbosity $ "Trying to locate mirrors via DNS for initial bootstrap of secure repository (" ++ show remoteRepoURI ++ ") ..."
  Sec.DNS.queryBootstrapMirrors verbosity remoteRepoURI
else ...
```

?
```